staging/src/k8s.io/kubectl/pkg/cmd/delete/delete.go

```diff
@@ -161,16 +160,29 @@ func (o *DeleteOptions) Complete(f cmdutil.Factory, args []string, cmd *cobra.Co
 			o.IgnoreNotFound = true
 		}
 	}
-	if o.DeleteNow {
-		if o.GracePeriod != -1 {
-			return fmt.Errorf("--now and --grace-period cannot be specified together")
+
+	if o.ForceDeletion {
+		// It is invalid to specify both the force flag and a grace period (other than 0), so fail in that case
+		if o.GracePeriod != -1 && o.GracePeriod != 0 {
+			return fmt.Errorf("--force and --grace-period greater than 0 cannot be specified together")
+		}
+		// Force is really just grace period of 0
+		o.GracePeriod = 0
+	} else {
+		if o.DeleteNow {
+			// It is invalid to specify both the delete now flag and a grace period, so fail in that case
+			if o.GracePeriod != -1 {
+				return fmt.Errorf("--now and --grace-period cannot be specified together")
+			}
+			// DeleteNow is really just grace period of 1
+			o.GracePeriod = 1
+		}
+		if o.GracePeriod == 0 {
+			// Grace period of 0 actually causes a force deletion, which is probably not what the user intended if they
+			// did not explicitly use the force flag, so convert grace period of 0 into grace period of 1 to prevent
+			// accidental data loss.
+			o.GracePeriod = 1
```

Comment:
This nested ifs complicate the flow a lot and mixes filling the internal structure (`Complete` method) with validation (`Validate` method), I'd suggest just add:
```go
if o.ForceDeletion && o.GracePeriod <= 0 {
    o.GracePeriod = 0
}
```
after the old conditions.

Reply:
Agreed.  This is a much cleaner change.  Thanks